Ebean entity enhancement failing

I've tried the Play 2.4.0-RC1 migration guide against the [computer-database-java](https://github.com/typesafehub/activator-computer-database-java) (an activator template) and found that entity enhancement is broken.

## How to reproduce:

1. Clone the following fork of computer-database-java: https://github.com/marcospereira/activator-computer-database-java
2. run `activator run`
3. Try to access http://localhost:9000/
4. You will see the following stack trace:

```
CreationException: Guice creation errors:

1) Error injecting constructor, java.lang.IllegalStateException: Bean class models.Category is not enhanced?
  at play.db.ebean.EbeanDynamicEvolutions.<init>(Unknown Source)
  at play.db.ebean.EbeanDynamicEvolutions.class(Unknown Source)
  while locating play.db.ebean.EbeanDynamicEvolutions
  at play.db.ebean.EbeanModule.bindings(EbeanModule.java:23):
Binding(class play.api.db.evolutions.DynamicEvolutions to ConstructionTarget(class play.db.ebean.EbeanDynamicEvolutions) eagerly)
  while locating play.api.db.evolutions.DynamicEvolutions
  while locating play.api.db.evolutions.ApplicationEvolutionsProvider
  at play.api.db.evolutions.EvolutionsModule.bindings(EvolutionsModule.scala:22):
Binding(class play.api.db.evolutions.ApplicationEvolutions to ProviderConstructionTarget(class play.api.db.evolutions.ApplicationEvolutionsProvider) eagerly)
  while locating play.api.db.evolutions.ApplicationEvolutions
Caused by: java.lang.IllegalStateException: Bean class models.Category is not enhanced?
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.setEntityBeanClass(BeanDescriptorManager.java:1365)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.createByteCode(BeanDescriptorManager.java:1103)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.readDeployAssociations(BeanDescriptorManager.java:1023)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.readEntityDeploymentAssociations(BeanDescriptorManager.java:502)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.deploy(BeanDescriptorManager.java:219)
	at com.avaje.ebeaninternal.server.core.InternalConfiguration.<init>(InternalConfiguration.java:119)
	at com.avaje.ebeaninternal.server.core.DefaultServerFactory.createServer(DefaultServerFactory.java:142)
	at com.avaje.ebeaninternal.server.core.DefaultServerFactory.createServer(DefaultServerFactory.java:38)
	at com.avaje.ebean.EbeanServerFactory.createInternal(EbeanServerFactory.java:79)
	at com.avaje.ebean.EbeanServerFactory.create(EbeanServerFactory.java:64)
	at play.db.ebean.EbeanDynamicEvolutions.start(EbeanDynamicEvolutions.java:52)
	at play.db.ebean.EbeanDynamicEvolutions.<init>(EbeanDynamicEvolutions.java:38)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at com.google.inject.internal.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:85)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.Scopes$1$1.get(Scopes.java:65)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.Scopes$1$1.get(Scopes.java:65)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.InjectorImpl$4$1.call(InjectorImpl.java:978)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.InjectorImpl$4.get(InjectorImpl.java:974)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1013)
	at play.api.inject.guice.GuiceInjector.instanceOf(GuiceInjectorBuilder.scala:314)
	at play.api.inject.guice.GuiceInjector.instanceOf(GuiceInjectorBuilder.scala:309)
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get$lzycompute(EvolutionsModule.scala:53)
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:53)
	at play.api.db.evolutions.ApplicationEvolutionsProvider.get(EvolutionsModule.scala:44)
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:55)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.Scopes$1$1.get(Scopes.java:65)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:204)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:198)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:198)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:179)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
	at com.google.inject.Guice.createInjector(Guice.java:95)
	at com.google.inject.Guice.createInjector(Guice.java:72)
	at com.google.inject.Guice.createInjector(Guice.java:62)
	at play.api.inject.guice.GuiceBuilder.createInjector(GuiceInjectorBuilder.scala:119)
	at play.api.inject.guice.GuiceApplicationBuilder.injector(GuiceApplicationBuilder.scala:87)
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:93)
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:19)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:153)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:150)
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:150)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:121)
	at scala.Option.map(Option.scala:146)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:121)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:119)
	at scala.util.Success.flatMap(Try.scala:230)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:119)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:111)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1689)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

2) Error injecting constructor, java.lang.IllegalStateException: Bean class models.Category is not enhanced?
  at play.db.ebean.EbeanDynamicEvolutions.<init>(Unknown Source)
  at play.db.ebean.EbeanDynamicEvolutions.class(Unknown Source)
  while locating play.db.ebean.EbeanDynamicEvolutions
  at play.db.ebean.EbeanModule.bindings(EbeanModule.java:23):
Binding(class play.api.db.evolutions.DynamicEvolutions to ConstructionTarget(class play.db.ebean.EbeanDynamicEvolutions) eagerly)
  while locating play.api.db.evolutions.DynamicEvolutions
Caused by: java.lang.IllegalStateException: Bean class models.Category is not enhanced?
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.setEntityBeanClass(BeanDescriptorManager.java:1365)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.createByteCode(BeanDescriptorManager.java:1103)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.readDeployAssociations(BeanDescriptorManager.java:1023)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.readEntityDeploymentAssociations(BeanDescriptorManager.java:502)
	at com.avaje.ebeaninternal.server.deploy.BeanDescriptorManager.deploy(BeanDescriptorManager.java:219)
	at com.avaje.ebeaninternal.server.core.InternalConfiguration.<init>(InternalConfiguration.java:119)
	at com.avaje.ebeaninternal.server.core.DefaultServerFactory.createServer(DefaultServerFactory.java:142)
	at com.avaje.ebeaninternal.server.core.DefaultServerFactory.createServer(DefaultServerFactory.java:38)
	at com.avaje.ebean.EbeanServerFactory.createInternal(EbeanServerFactory.java:79)
	at com.avaje.ebean.EbeanServerFactory.create(EbeanServerFactory.java:64)
	at play.db.ebean.EbeanDynamicEvolutions.start(EbeanDynamicEvolutions.java:52)
	at play.db.ebean.EbeanDynamicEvolutions.<init>(EbeanDynamicEvolutions.java:38)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at com.google.inject.internal.DefaultConstructionProxyFactory$1.newInstance(DefaultConstructionProxyFactory.java:85)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:254)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.Scopes$1$1.get(Scopes.java:65)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:54)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1031)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.Scopes$1$1.get(Scopes.java:65)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:40)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:204)
	at com.google.inject.internal.InternalInjectorCreator$1.call(InternalInjectorCreator.java:198)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1024)
	at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:198)
	at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:179)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:109)
	at com.google.inject.Guice.createInjector(Guice.java:95)
	at com.google.inject.Guice.createInjector(Guice.java:72)
	at com.google.inject.Guice.createInjector(Guice.java:62)
	at play.api.inject.guice.GuiceBuilder.createInjector(GuiceInjectorBuilder.scala:119)
	at play.api.inject.guice.GuiceApplicationBuilder.injector(GuiceApplicationBuilder.scala:87)
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:93)
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:19)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:153)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1$$anonfun$2.apply(DevServerStart.scala:150)
	at play.utils.Threads$.withContextClassLoader(Threads.scala:21)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:150)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1$$anonfun$1.apply(DevServerStart.scala:121)
	at scala.Option.map(Option.scala:146)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:121)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1$$anonfun$apply$1.apply(DevServerStart.scala:119)
	at scala.util.Success.flatMap(Try.scala:230)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:119)
	at play.core.server.DevServerStart$$anonfun$mainDev$1$$anon$1$$anonfun$get$1.apply(DevServerStart.scala:111)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
	at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1689)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)

2 errors
```

--- a/build.sbt
+++ b/build.sbt
@@ -146,7 +146,8 @@ def noPublish: Seq[Setting[_]] = Seq(
 def playEbeanDeps = Seq(
   "com.typesafe.play" %% "play-java-jdbc" % PlayVersion,
   "org.avaje.ebeanorm" % "avaje-ebeanorm" % "4.5.6",
-  avajeEbeanormAgent
+  avajeEbeanormAgent,
+  "com.typesafe.play" %% "play-test" % PlayVersion % Test
 )
 
 def sbtPlayEbeanDeps = Seq(

--- a/play-ebean/src/main/java/play/db/ebean/DefaultEbeanConfig.java
+++ b/play-ebean/src/main/java/play/db/ebean/DefaultEbeanConfig.java
@@ -4,9 +4,6 @@
 package play.db.ebean;
 
 import com.avaje.ebean.config.ServerConfig;
-import com.typesafe.config.Config;
-import com.typesafe.config.ConfigValue;
-import com.typesafe.config.ConfigValueType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import play.Configuration;
@@ -68,71 +65,55 @@ public class DefaultEbeanConfig implements EbeanConfig {
          */
         public EbeanConfig parse() {
 
-            Config config = configuration.underlying();
-            Config playEbeanConfig = config.getConfig("play.ebean");
-            String defaultServer = playEbeanConfig.getString("defaultDatasource");
-            String ebeanConfigKey = playEbeanConfig.getString("config");
+            EbeanParsedConfig config = EbeanParsedConfig.parseFromConfig(configuration);
 
             Map<String, ServerConfig> serverConfigs = new HashMap<String, ServerConfig>();
 
-            if (config.hasPath(ebeanConfigKey)) {
-                Config ebeanConfig = config.getConfig(ebeanConfigKey);
-                Configuration conf = new Configuration(ebeanConfig);
+            for (Map.Entry<String, List<String>> entry: config.getDatasourceModels().entrySet()) {
+                String key = entry.getKey();
 
-                for (String key: ebeanConfig.root().keySet()) {
+                ServerConfig serverConfig = new ServerConfig();
+                serverConfig.setName(key);
+                serverConfig.loadFromProperties();
+                try {
+                    serverConfig.setDataSource(new WrappingDatasource(dbApi.getDatabase(key).getDataSource()));
+                } catch(Exception e) {
+                    throw configuration.reportError(
+                        "ebean." + key,
+                        e.getMessage(),
+                        e
+                    );
+                }
 
-                    ServerConfig serverConfig = new ServerConfig();
-                    serverConfig.setName(key);
-                    serverConfig.loadFromProperties();
+                if (config.getDefaultDatasource().equals(key)) {
+                    serverConfig.setDefaultServer(true);
+                }
+
+                Set<String> classes = new HashSet<String>();
+                for (String load: entry.getValue()) {
+                    load = load.trim();
+                    if (load.endsWith(".*")) {
+                        classes.addAll(play.libs.Classpath.getTypes(environment, load.substring(0, load.length()-2)));
+                    } else {
+                        classes.add(load);
+                    }
+                }
+                for (String clazz: classes) {
                     try {
-                        serverConfig.setDataSource(new WrappingDatasource(dbApi.getDatabase(key).getDataSource()));
-                    } catch(Exception e) {
-                        throw conf.reportError(
-                            key,
-                            e.getMessage(),
+                        serverConfig.addClass(Class.forName(clazz, true, environment.classLoader()));
+                    } catch (Throwable e) {
+                        throw configuration.reportError(
+                            "ebean." + key,
+                            "Cannot register class [" + clazz + "] in Ebean server",
                             e
                         );
                     }
-
-                    if (defaultServer.equals(key)) {
-                        serverConfig.setDefaultServer(true);
-                    }
-
-                    ConfigValue raw = ebeanConfig.getValue(key);
-                    List<String> toLoad;
-                    if (raw.valueType() == ConfigValueType.STRING) {
-                        // Support legacy comma separated string
-                        toLoad = Arrays.asList(((String) raw.unwrapped()).split(","));
-                    } else {
-                        toLoad = ebeanConfig.getStringList(key);
-                    }
-
-                    Set<String> classes = new HashSet<String>();
-                    for (String load: toLoad) {
-                        load = load.trim();
-                        if (load.endsWith(".*")) {
-                            classes.addAll(play.libs.Classpath.getTypes(environment, load.substring(0, load.length()-2)));
-                        } else {
-                            classes.add(load);
-                        }
-                    }
-                    for (String clazz: classes) {
-                        try {
-                            serverConfig.addClass(Class.forName(clazz, true, environment.classLoader()));
-                        } catch (Throwable e) {
-                            throw conf.reportError(
-                                key,
-                                "Cannot register class [" + clazz + "] in Ebean server",
-                                e
-                            );
-                        }
-                    }
-
-                    serverConfigs.put(key, serverConfig);
                 }
+
+                serverConfigs.put(key, serverConfig);
             }
 
-            return new DefaultEbeanConfig(defaultServer, serverConfigs);
+            return new DefaultEbeanConfig(config.getDefaultDatasource(), serverConfigs);
         }
 
         /**

--- a/play-ebean/src/main/java/play/db/ebean/EbeanParsedConfig.java
+++ b/play-ebean/src/main/java/play/db/ebean/EbeanParsedConfig.java
@@ -1,0 +1,66 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package play.db.ebean;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValue;
+import com.typesafe.config.ConfigValueType;
+import play.Configuration;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The raw parsed config from Ebean, as opposed to the EbeanConfig which actually requires starting database connection
+ * pools to create.
+ */
+public class EbeanParsedConfig {
+
+    private final String defaultDatasource;
+    private final Map<String, List<String>> datasourceModels;
+
+    public EbeanParsedConfig(String defaultDatasource, Map<String, List<String>> datasourceModels) {
+        this.defaultDatasource = defaultDatasource;
+        this.datasourceModels = datasourceModels;
+    }
+
+    public String getDefaultDatasource() {
+        return defaultDatasource;
+    }
+
+    public Map<String, List<String>> getDatasourceModels() {
+        return datasourceModels;
+    }
+
+    public static EbeanParsedConfig parseFromConfig(Configuration configuration) {
+        Config config = configuration.underlying();
+        Config playEbeanConfig = config.getConfig("play.ebean");
+        String defaultDatasource = playEbeanConfig.getString("defaultDatasource");
+        String ebeanConfigKey = playEbeanConfig.getString("config");
+
+        Map<String, List<String>> datasourceModels = new HashMap<>();
+
+        if (config.hasPath(ebeanConfigKey)) {
+            Config ebeanConfig = config.getConfig(ebeanConfigKey);
+
+            for (String key : ebeanConfig.root().keySet()) {
+
+                ConfigValue raw = ebeanConfig.getValue(key);
+                List<String> models;
+                if (raw.valueType() == ConfigValueType.STRING) {
+                    // Support legacy comma separated string
+                    models = Arrays.asList(((String) raw.unwrapped()).split(","));
+                } else {
+                    models = ebeanConfig.getStringList(key);
+                }
+
+                datasourceModels.put(key, models);
+
+            }
+        }
+        return new EbeanParsedConfig(defaultDatasource, datasourceModels);
+    }
+}

--- a/play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java
+++ b/play-ebean/src/main/java/play/db/ebean/ModelsConfigLoader.java
@@ -1,0 +1,25 @@
+package play.db.ebean;
+
+import play.Configuration;
+import play.Environment;
+import play.Mode;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Given a classloader, load the models configuration.
+ *
+ * This is used by the ebean sbt plugin to get the same models configuration that will be loaded by the app.
+ */
+public class ModelsConfigLoader implements Function<ClassLoader, Map<String, List<String>>> {
+
+    @Override
+    public  Map<String, List<String>> apply(ClassLoader classLoader) {
+        Environment env = new Environment(new File("."), classLoader, Mode.PROD);
+        Configuration config = Configuration.load(env);
+        return EbeanParsedConfig.parseFromConfig(config).getDatasourceModels();
+    }
+}

--- a/play-ebean/src/test/java/play/db/ebean/EbeanParsedConfigTest.java
+++ b/play-ebean/src/test/java/play/db/ebean/EbeanParsedConfigTest.java
@@ -1,0 +1,73 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package play.db.ebean;
+
+import com.google.common.collect.ImmutableMap;
+import com.typesafe.config.ConfigFactory;
+import play.Configuration;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+import org.junit.*;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+
+public class EbeanParsedConfigTest {
+
+    private EbeanParsedConfig parse(Map<String, ? extends Object> config) {
+        return EbeanParsedConfig.parseFromConfig(
+                new Configuration(ConfigFactory.parseMap(config).withFallback(ConfigFactory.defaultReference()))
+        );
+    }
+
+    @Test
+    public void defaultConfig() {
+        EbeanParsedConfig config = parse(Collections.emptyMap());
+        assertThat(config.getDefaultDatasource(), equalTo("default"));
+        assertThat(config.getDatasourceModels().size(), equalTo(0));
+    }
+
+    @Test
+    public void withDatasources() {
+        EbeanParsedConfig config = parse(ImmutableMap.of(
+                "ebean.default", Arrays.asList("a", "b"),
+                "ebean.other", Arrays.asList("c")
+        ));
+        assertThat(config.getDatasourceModels().size(), equalTo(2));
+        assertThat(config.getDatasourceModels().get("default"), hasItems("a", "b"));
+        assertThat(config.getDatasourceModels().get("other"), hasItems("c"));
+    }
+
+    @Test
+    public void commaSeparatedModels() {
+        EbeanParsedConfig config = parse(ImmutableMap.of(
+                "ebean.default", "a,b"
+        ));
+        assertThat(config.getDatasourceModels().get("default"), hasItems("a", "b"));
+    }
+
+    @Test
+    public void customDefault() {
+        EbeanParsedConfig config = parse(ImmutableMap.of(
+                "play.ebean.defaultDatasource", "custom"
+        ));
+        assertThat(config.getDefaultDatasource(), equalTo("custom"));
+    }
+
+    @Test
+    public void customConfig() {
+        EbeanParsedConfig config = parse(ImmutableMap.of(
+                "play.ebean.config", "my.custom",
+                "my.custom.default", Arrays.asList("a")
+        ));
+        assertThat(config.getDatasourceModels().size(), equalTo(1));
+        assertThat(config.getDatasourceModels().get("default"), hasItems("a"));
+    }
+
+
+}

--- a/sbt-play-ebean/src/main/scala/play/ebean/sbt/SbtEbean.scala
+++ b/sbt-play-ebean/src/main/scala/play/ebean/sbt/SbtEbean.scala
@@ -1,5 +1,7 @@
 package play.ebean.sbt
 
+import java.net.URLClassLoader
+
 import com.typesafe.play.sbt.enhancer.PlayEnhancer
 import sbt.Keys._
 import sbt._
@@ -99,24 +101,30 @@ object SbtEbean extends AutoPlugin {
 
 
   private def configuredEbeanModels = Def.task {
-    import com.typesafe.config._
     import collection.JavaConverters._
+    import java.util.{ Map => JMap, List => JList }
 
-    val configFile = sys.props.get("config.file").map(new File(_)).orElse {
-      val configResource = sys.props.get("config.resource").getOrElse("application.conf")
-      unmanagedResources.value.filter(r => r.exists && !r.isDirectory)
-        .pair(relativeTo(unmanagedResourceDirectories.value))
-        .collectFirst {
-          case (file, name) if name == configResource => file
-        }
+    // Creates a classloader with all the dependencies and all the resources, from there we can use the play ebean
+    // code to load the config as it would be loaded in production
+    def withClassLoader[T](block: ClassLoader => T): T = {
+      val classpath = unmanagedResourceDirectories.value.map(_.toURI.toURL) ++ dependencyClasspath.value.map(_.data.toURI.toURL)
+      val classLoader = new URLClassLoader(classpath.toArray, null)
+      try {
+        block(classLoader)
+      } finally {
+        classLoader.close()
+      }
     }
 
-    configFile.fold(Seq("models.*")) { file =>
-      val config = ConfigFactory.parseFileAnySyntax(file)
-      try {
-        config.getConfig("ebean").entrySet.asScala.map(_.getValue.unwrapped.toString).toSeq.distinct
-      } catch {
-        case e: ConfigException.Missing => Seq("models.*")
+    withClassLoader { classLoader =>
+      val configLoader = classLoader.loadClass("play.db.ebean.ModelsConfigLoader").
+        asSubclass(classOf[java.util.function.Function[ClassLoader, JMap[String, JList[String]]]])
+      val config = configLoader.newInstance().apply(classLoader)
+
+      if (config.isEmpty) {
+        Seq("models.*")
+      } else {
+        config.asScala.flatMap(_._2.asScala).toSeq.distinct
       }
     }
   }

--- a/sbt-play-ebean/src/sbt-test/sbt-ebean/enhancement/conf/application.conf
+++ b/sbt-play-ebean/src/sbt-test/sbt-ebean/enhancement/conf/application.conf
@@ -1,1 +1,1 @@
-ebean.default = "models.*"
+ebean.default = [ "models.*" ]

